pkg/app/app.go

```diff
@@ -193,23 +217,43 @@ func (app *App) setupRepo() (bool, error) {
 		}
 
 		if !shouldInitRepo {
-			// check if we have a recent repo we can open
-			for _, repoDir := range app.Config.GetAppState().RecentRepos {
-				if isRepo, _ := isDirectoryAGitRepository(repoDir); isRepo {
-					if err := os.Chdir(repoDir); err == nil {
-						return true, nil
-					}
-				}
+			// Attempt to open a recent repo, exit if no repo could be opened
+			if didOpenRepo := openRecentRepo(app); !didOpenRepo {
+				fmt.Println(app.Tr.NoRecentRepositories)
+				os.Exit(1)
 			}
 
-			fmt.Println(app.Tr.NoRecentRepositories)
-			os.Exit(1)
+			return true, nil
 		}
 		if err := app.OSCommand.Cmd.New("git init " + initialBranch).Run(); err != nil {
 			return false, err
 		}
 	}
 
+	// Run this afterward so that the previous repo creation steps can run without this interfering
+	if isBare, err := isBareRepo(app.OSCommand); isBare {
+		if err != nil {
+			return false, err
+		}
+
+		fmt.Print(app.Tr.BareRepo)
+
+		response, _ := bufio.NewReader(os.Stdin).ReadString('\n')
+		shouldOpenRecent := strings.Trim(response, " \r\n") == "y"
```

Comment:
This definition can be moved to the if below, just like with `didOpenRepo` on line `221`.